### PR TITLE
Various database migration fixes

### PR DESCRIPTION
**Pull request by DeepDiver1975**
- sqlite int vs bigint for autoincrement
- oracle autoincrement and unsigned
- all var char vs clob

### CODE REVIEW

**Comment by PVince81** [comment]
@butonic I heard you love Oracle stuff

File: lib/private/DB/Migrator.php [hunk 1]
```diff
@@ -194,7 +196,25 @@ protected function renameTableSchema(Table $table, $newName) {
...
+	 * @param \Doctrine\DBAL\Connection $connection
+	 * @return \Doctrine\DBAL\Schema\SchemaDiff
+	 * @throws DBALException
+	 */
 	protected function getDiff(Schema $targetSchema, \Doctrine\DBAL\Connection $connection) {
+		// adjust varchar columns with a length higher then getVarcharMaxLength to clob
+		foreach ($targetSchema->getTables() as $table) {
+			foreach ($table->getColumns() as $column) {
+				if ($column->getType() instanceof StringType) {
+					if ($column->getLength() > $connection->getDatabasePlatform()->getVarcharMaxLength()) {
```
Comment (butonic) [inline lib/private/DB/Migrator.php]: ok, with this you make the type conversion magic more explicit. 👍

File: lib/private/DB/OracleMigrator.php [hunk 2]
```diff
@@ -39,7 +40,12 @@ protected function getDiff(Schema $targetSchema, \Doctrine\DBAL\Connection $conn
 			$tableDiff->name = $this->connection->quoteIdentifier($tableDiff->name);
 			foreach ($tableDiff->changedColumns as $column) {
 				$column->oldColumnName = $this->connection->quoteIdentifier($column->oldColumnName);
+				// auto increment is not relevant for oracle and can anyhow not be applied on change
+				$column->changedProperties = array_diff($column->changedProperties, ['autoincrement', 'unsigned']);
```
Comment (butonic) [inline lib/private/DB/OracleMigrator.php]: yes, autoincrement and unsigned will never be read back from oracle 👍

File: tests/lib/DB/SchemaDiffTest.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+		// no sql statement is expected
+		$this->assertEquals([], $diff->toSql($this->connection->getDatabasePlatform()));
+	}
+
+	public function providesSchemaFiles() {
+		return [
+			'explicit test on autoincrement' => [file_get_contents(__DIR__ . '/schemDiffData/autoincrement.xml')],
+			'explicit test on clob' => [file_get_contents(__DIR__ . '/schemDiffData/clob.xml')],
+			'explicit test on unsigned' => [file_get_contents(__DIR__ . '/schemDiffData/unsigned.xml')],
+			'testing core schema' => [file_get_contents(__DIR__ . '/schemDiffData/core.xml')],
```
Comment (butonic) [inline tests/lib/DB/SchemaDiffTest.php]: hm ... better test the db_schema.xml? ... nobody will update the core.xml when db_schema.xml changes

Comment (DeepDiver1975) [inline tests/lib/DB/SchemaDiffTest.php]: there are two issues
- index names are causing conflicts
- core.xml takes pretty long to execute - I prefer to have explicit small xmls ... so this file will die once this is clean on all dbs

**Comment by DeepDiver1975** [comment]
:see_no_evil: pg has issues as well

**Comment by DeepDiver1975** [comment]
Nice - killed my own fix fro pg :cry:

**Comment by DeepDiver1975** [comment]
@PVince81 @icewind1991 @butonic @dragotin @guruz please review - THX

**Comment by PVince81** [comment]
@DeepDiver1975 pgsql test timed out / too many connections.

Can you run it locally, do the tests pass ?

**Comment by DeepDiver1975** [comment]
> Can you run it locally, do the tests pass ?

yes - they pass - before squashing they have been green already

File: lib/private/DB/PostgreSqlMigrator.php [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+	/**
+	 * @param Schema $targetSchema
+	 * @param \Doctrine\DBAL\Connection $connection
+	 * @return \Doctrine\DBAL\Schema\SchemaDiff
+	 */
+	protected function getDiff(Schema $targetSchema, \Doctrine\DBAL\Connection $connection) {
+		$schemaDiff = parent::getDiff($targetSchema, $connection);
+
+		foreach ($schemaDiff->changedTables as $tableDiff) {
+			// fix default value in brackets
```
Comment (PVince81) [inline lib/private/DB/PostgreSqlMigrator.php]: maybe add a hint why this is needed in the first place ?

From reading the code I can't really tell

**Comment by PVince81** [comment]
The code looks good :+1: 

I can't judge whether this all makes sense but I trust that there is a good reason for it and also that the unit tests will show that this all works.

Tried a local upgrade from v9.0.2 (git) to this branch with sqlite and myself and there was no issue.

**Comment by icewind1991** [comment]
Looks good :+1:

**Comment by butonic** [comment]
👍 see comments above, and new tests are passing on CI

**Pull request merged**